test/e2e/pd.go

```diff
@@ -50,10 +51,15 @@ var _ = Describe("PD", func() {
 
 		nodes, err := c.Nodes().List(labels.Everything(), fields.Everything())
 		expectNoError(err, "Failed to list nodes for e2e cluster.")
-		Expect(len(nodes.Items) >= 2).To(BeTrue())
 
-		host0Name = nodes.Items[0].ObjectMeta.Name
-		host1Name = nodes.Items[1].ObjectMeta.Name
+		numHosts = len(nodes.Items)
+
+		if len(nodes.Items) >= 2 {
+			host1Name = nodes.Items[1].ObjectMeta.Name
+		}
+		if len(nodes.Items) >= 1 {
```

Comment:
nit: Since the code below always assumes >= 2 hosts, I'm not sure why we'd do a separate if check here for an e2e cluster with only 1 host. This could just be done inside the prior if block.